pkg/arvo/lib/versioning.hoon

```diff
@@ -0,0 +1,52 @@
+/+  agentio
+|_  [=bowl:gall root=mark version=@ud min=@ud]
++*  io  ~(. agentio bowl)
+++  is-root
+  |=  =mark
+  ?~  (rush mark mark-parser)
+    %.n
+  %.y
+::
+++  mark-parser
+  ;~(pfix (jest root) ;~(pose ;~(pfix hep dum:ag) (easy `@ud`0)))
+::
+++  read-version
+  |=  =mark
+  (rash mark mark-parser)
+::
+++  append-version
+  |=  ver=@ud
+  :((cury cat 3) root '-' (scot %ud ver))
+::
+++  current-version
+  (append-version version)
+::
+++  supported
+  |=  =mark
+  =/  ver
+    (read-version mark)
+  &((gte ver min) (lte ver version))
+::
+++  convert-to
+  |=  =cage
+  ^-  vase
+  ?:  =(p.cage current-version)
+    q.cage
+  ((tube-to p.cage) q.cage)
```

Comment:
Are we planning on building out any sort of caching utility for this? (Honestly, starting to think we should build a "ford-hook" (forgive the horrible name) that lets one start a subscription to a file or tube with a poke to the app and optionally subscribe to the results. That way all the caching logic for mark conversions can live in one app, and all of the state for "cached marks" can exist together.

Reply:
As of #4491 you can achieve the same by importing the relevant marks into your app I believe. You can't get _as_ dynamic with it, but I don't think you want/need to.

Reply:
I think we'd like to retain our current dynamic behavior with `%graph-validator` marks, at least. For normal "update versions" perhaps it would make the most sense to do it statically.